x/wasm/keeper/relay.go

```diff
@@ -130,13 +133,34 @@ func (k Keeper) OnRecvPacket(
 	res, gasUsed, execErr := k.wasmVM.IBCPacketReceive(codeInfo.CodeHash, env, msg, prefixStore, cosmwasmAPI, querier, ctx.GasMeter(), gas, costJSONDeserialization)
 	k.consumeRuntimeGas(ctx, gasUsed)
 	if execErr != nil {
-		panic(execErr)
+		panic(execErr) // let contract fully abort IBC receive in certain case
```

Comment:
"in certain case" is a bit vague.

Good to link to some doc (in another repo?) when `IBCPacketReceive` returns `err != nil` and when it returns `err == nil && res.Err != nil`.

Reply:
Reading for the impact of the change, I believe what actually changes here (besides improvements with emitting events) was to be the "abort tx when contract panics". Which is detected by `execErr != nil`, right?

So this worked the same before and after this PR???

Reply:
This PR does not show the changes to v0.31 correct. It was this before:
```go
res, gasUsed, execErr := k.wasmVM.IBCPacketReceive(codeInfo.CodeHash, env, msg, prefixStore, cosmwasmAPI, querier, ctx.GasMeter(), gas, costJSONDeserialization)
k.consumeRuntimeGas(ctx, gasUsed)
if execErr != nil {
	return nil, sdkerrors.Wrap(types.ErrExecuteFailed, execErr.Error())
}
if res.Err != "" { // handle error case as before https://github.com/CosmWasm/wasmvm/commit/c300106fe5c9426a495f8e10821e00a9330c56c6
	return nil, sdkerrors.Wrap(types.ErrExecuteFailed, res.Err)
}
```

https://github.com/CosmWasm/wasmd/blob/v0.31.0/x/wasm/keeper/relay.go#L130-L137